### PR TITLE
[DoctrineCodeQuality] replace `ASC`|`DESC` strings with constants

### PR DESCRIPTION
**Pull request by vladyslavstartsev**
As discussed here https://github.com/rectorphp/rector/issues/5183#issuecomment-760530050

### CODE REVIEW

File: config/set/doctrine-code-quality.php [hunk 1]
```diff
@@ -26,4 +31,17 @@
...
     $services->set(RemoveRedundantDefaultClassAnnotationValuesRector::class);
+    $services->set(ReplaceStringWithClassConstantRector::class)
+        ->call('configure', [
+            [
+                ReplaceStringWithClassConstantRector::REPLACE_STRING_WITH_CLASS_CONSTANT => ValueObjectInliner::inline([
+                    new ReplaceStringWithClassConstant(
+                        QueryBuilder::class,
+                        'orderBy',
+                        1,
+                        Criteria::class
```
Comment (TomasVotruba) [inline config/set/doctrine-code-quality.php:42]: These 2 classes must be defined as string, as they don't exist in Rector.

**Comment by vladyslavstartsev** [comment]
@TomasVotruba 

any idea how to convert this?

```diff
- ->findOneBy(['id' => 1], ['dateEnd' => 'DESC']);
+ ->findOneBy(['id' => 1], ['dateEnd' => Criteria::DESC])
```
or
```diff
- ->findOneBy(['lastName' => 'Smith'], ['dateEnd' => 'DESC']);
+ ->findOneBy(['lastName' => 'Smith'], ['dateEnd' => Criteria::DESC])
```

**Comment by TomasVotruba** [comment]
Thanks :+1:

**Comment by TomasVotruba** [comment]
@vladyslavstartsev The `ReplaceStringWithClassConstantRector` must be extended to work with nested arrays as well.
With use of `traverseWithCallable()` method.

**Pull request merged**